jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpURI.java

```diff
@@ -1452,6 +1457,9 @@ else if (c == '/')
                 }
             }
 
+            if (last != null && state.ordinal() > last.ordinal())
+                throw new IllegalArgumentException("uri cannot go beyond " + last);
+
```

Comment:
I don't think we should not parse past the fragment etc.  We should parse to the end, but record a violation if there is a fragment.